docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -69,25 +104,24 @@ A list to a few of the popular ones:
 
 **Examples**:
 
-- IntelliJ IDEA introduced an inspection rule for Java 17 or higher to detect
+- **Check updates**: IntelliJ IDEA introduced an inspection rule for Java 17 or higher to detect
   redundant `strictfp` modifiers. This resulted in the analysis of `RedundantModifierCheck`
   to verify if we needed to update it.
 
-### Real Usage Examples in Large Projects
+- **New check**: Sonar introduced a [new rule](https://rules.sonarsource.com/java/tag/java21/RSPEC-6916/)
+  to suggest the use of `when` instead of a single if statement inside
+  a pattern match body. This resulted in the creation
+  of a new check `WhenShouldBeUsed` to enforce the use of this new feature.
+
+### 1.5 Real Usage Examples in Large Projects
 
 Review real usage examples of the new language feature in large projects.
 This helps to identify potential issues and ensures that the checks
 are aligned with practical use cases.
 A list of representative projects can be found
 [here](https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/github-action-projects1.properties).
```

Comment:
I am curious why we pointed to this file, when we made updates to the other file only.
https://github.com/checkstyle/contribution/commit/7d57a4f06df2a11cb311bddf7e14a8910cc1c13d

Reply:
We only made updates to the files used in the regression. The file mentioned in this document is not used by any of our workflows or report generation  (https://github.com/checkstyle/contribution/issues/880) that's why we didn't update it.

Should we just update it to keep new projects documented somewhere in a separate file?

Reply:
Since the mentioned issue is leading to removing the files, I am not sure I see the point to updating it unless we find a reason for the files to continue to exist. I still see us just pointing to https://github.com/checkstyle/contribution/commit/7d57a4f06df2a11cb311bddf7e14a8910cc1c13d . If we want to identify the newer 21+ projects in that, then we can do that.

@nrmancuso

Reply:
> unless we find a reason for the files to continue to exist

maybe a reason is "documenting new project links"

as we can't add all of them in the regression project

Reply:
We can just link to the folder where project files live, otherwise this is bound bound to become outdated

Reply:
link to the folder
 
 > A list of representative projects can be found [here](https://github.com/checkstyle/contribution/tree/master/checkstyle-tester) in projects files.

Reply:
linking to the folder is a good idea
 
 > A list of representative projects can be found [here](https://github.com/checkstyle/contribution/tree/master/checkstyle-tester) in projects files.